2018-02-03 05:33:22,740 ERROR: Exception on /index [GET] [in c:\lren\projects\microblog\venv\lib\site-packages\flask\app.py:1560]
Traceback (most recent call last):
  File "c:\lren\projects\microblog\venv\lib\site-packages\flask\app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "c:\lren\projects\microblog\venv\lib\site-packages\flask\app.py", line 1614, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "c:\lren\projects\microblog\venv\lib\site-packages\flask\app.py", line 1517, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "c:\lren\projects\microblog\venv\lib\site-packages\flask\_compat.py", line 33, in reraise
    raise value
  File "c:\lren\projects\microblog\venv\lib\site-packages\flask\app.py", line 1612, in full_dispatch_request
    rv = self.dispatch_request()
  File "c:\lren\projects\microblog\venv\lib\site-packages\flask\app.py", line 1598, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "c:\lren\projects\microblog\venv\lib\site-packages\flask_login\utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "C:\lren\projects\microblog\app\routes.py", line 40, in index
    posts=posts)
  File "c:\lren\projects\microblog\venv\lib\site-packages\flask\templating.py", line 133, in render_template
    return _render(ctx.app.jinja_env.get_or_select_template(template_name_or_list),
  File "c:\lren\projects\microblog\venv\lib\site-packages\jinja2\environment.py", line 869, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "c:\lren\projects\microblog\venv\lib\site-packages\jinja2\environment.py", line 830, in get_template
    return self._load_template(name, self.make_globals(globals))
  File "c:\lren\projects\microblog\venv\lib\site-packages\jinja2\environment.py", line 804, in _load_template
    template = self.loader.load(self, name, globals)
  File "c:\lren\projects\microblog\venv\lib\site-packages\jinja2\loaders.py", line 125, in load
    code = environment.compile(source, name, filename)
  File "c:\lren\projects\microblog\venv\lib\site-packages\jinja2\environment.py", line 591, in compile
    self.handle_exception(exc_info, source_hint=source_hint)
  File "c:\lren\projects\microblog\venv\lib\site-packages\jinja2\environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "c:\lren\projects\microblog\venv\lib\site-packages\jinja2\_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "C:\lren\projects\microblog\app\templates\index.html", line 20, in template
    {% endif %}
  File "c:\lren\projects\microblog\venv\lib\site-packages\jinja2\environment.py", line 497, in _parse
    return Parser(self, source, name, encode_filename(filename)).parse()
  File "c:\lren\projects\microblog\venv\lib\site-packages\jinja2\parser.py", line 901, in parse
    result = nodes.Template(self.subparse(), lineno=1)
  File "c:\lren\projects\microblog\venv\lib\site-packages\jinja2\parser.py", line 883, in subparse
    rv = self.parse_statement()
  File "c:\lren\projects\microblog\venv\lib\site-packages\jinja2\parser.py", line 130, in parse_statement
    return getattr(self, 'parse_' + self.stream.current.value)()
  File "c:\lren\projects\microblog\venv\lib\site-packages\jinja2\parser.py", line 268, in parse_block
    node.body = self.parse_statements(('name:endblock',), drop_needle=True)
  File "c:\lren\projects\microblog\venv\lib\site-packages\jinja2\parser.py", line 165, in parse_statements
    result = self.subparse(end_tokens)
  File "c:\lren\projects\microblog\venv\lib\site-packages\jinja2\parser.py", line 883, in subparse
    rv = self.parse_statement()
  File "c:\lren\projects\microblog\venv\lib\site-packages\jinja2\parser.py", line 130, in parse_statement
    return getattr(self, 'parse_' + self.stream.current.value)()
  File "c:\lren\projects\microblog\venv\lib\site-packages\jinja2\parser.py", line 213, in parse_if
    'name:endif'))
  File "c:\lren\projects\microblog\venv\lib\site-packages\jinja2\parser.py", line 165, in parse_statements
    result = self.subparse(end_tokens)
  File "c:\lren\projects\microblog\venv\lib\site-packages\jinja2\parser.py", line 883, in subparse
    rv = self.parse_statement()
  File "c:\lren\projects\microblog\venv\lib\site-packages\jinja2\parser.py", line 130, in parse_statement
    return getattr(self, 'parse_' + self.stream.current.value)()
  File "c:\lren\projects\microblog\venv\lib\site-packages\jinja2\parser.py", line 199, in parse_for
    body = self.parse_statements(('name:endfor', 'name:else'))
  File "c:\lren\projects\microblog\venv\lib\site-packages\jinja2\parser.py", line 165, in parse_statements
    result = self.subparse(end_tokens)
  File "c:\lren\projects\microblog\venv\lib\site-packages\jinja2\parser.py", line 883, in subparse
    rv = self.parse_statement()
  File "c:\lren\projects\microblog\venv\lib\site-packages\jinja2\parser.py", line 144, in parse_statement
    self.fail_unknown_tag(token.value, token.lineno)
  File "c:\lren\projects\microblog\venv\lib\site-packages\jinja2\parser.py", line 97, in fail_unknown_tag
    return self._fail_ut_eof(name, self._end_token_stack, lineno)
  File "c:\lren\projects\microblog\venv\lib\site-packages\jinja2\parser.py", line 90, in _fail_ut_eof
    self.fail(' '.join(message), lineno)
  File "c:\lren\projects\microblog\venv\lib\site-packages\jinja2\parser.py", line 59, in fail
    raise exc(msg, lineno, self.name, self.filename)
jinja2.exceptions.TemplateSyntaxError: Encountered unknown tag 'endif'. You probably made a nesting mistake. Jinja is expecting this tag, but currently looking for 'endfor' or 'else'. The innermost block that needs to be closed is 'for'.
2018-02-03 05:33:29,636 ERROR: Exception on /explore [GET] [in c:\lren\projects\microblog\venv\lib\site-packages\flask\app.py:1560]
Traceback (most recent call last):
  File "c:\lren\projects\microblog\venv\lib\site-packages\flask\app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "c:\lren\projects\microblog\venv\lib\site-packages\flask\app.py", line 1614, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "c:\lren\projects\microblog\venv\lib\site-packages\flask\app.py", line 1517, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "c:\lren\projects\microblog\venv\lib\site-packages\flask\_compat.py", line 33, in reraise
    raise value
  File "c:\lren\projects\microblog\venv\lib\site-packages\flask\app.py", line 1612, in full_dispatch_request
    rv = self.dispatch_request()
  File "c:\lren\projects\microblog\venv\lib\site-packages\flask\app.py", line 1598, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "c:\lren\projects\microblog\venv\lib\site-packages\flask_login\utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "C:\lren\projects\microblog\app\routes.py", line 145, in explore
    posts = Post.query.ordery_by(Post.timestamp.desc()).all()
AttributeError: 'BaseQuery' object has no attribute 'ordery_by'
2018-02-03 05:34:40,277 INFO: Microblog startup [in C:\lren\projects\microblog\app\__init__.py:44]
